### PR TITLE
docs: Bulk spelling edits

### PR DESCRIPTION
**Pull request by ElPaisano**
Bulk edit for spelling. List of changes by page is below:

ARCHITECTURE.md
- All instances of "datastructure" / "datastructures" -> "data structure" / "data structures"
- System -> System, line 106
- apriori -> a priori, line 122

IMPORTERS_EXPORTERS.md
- All instances of "datastructure" / "datastructures" -> "data structure" / "data structures"
- All instances of "eg" -> "e.g." _(I realize this might be an English dialect thing, just suggesting the American English version because it's commonly used)_

IPIP/0001-lightweight-improvement-proposal-process.md
- Github -> GitHub, line 106

IPIP_PROCESS.md
- Github -> GitHub, line 78
- intial -> initial, line 88

KEYSTORE.md
- acommodate -> accommodate

MERKLE_DAG.md
- All instances of "datastructure" / "datastructures" -> "data structure" / "data structures"
- merkle tree -> Merkle tree, line 26 and line 86
- merkle dag -> merkledag, line 86 _(to match how it's written across the rest of the specs repo)_

REPO_FS.md
- All instances of "eg" -> "e.g."

http-gateways/PATH_GATEWAY.md
- remainer -> parameter, line 97 _(pretty sure this was a typo?)_
- provies -> proxies, line 353 _(pretty sure this was a typo?)_

http-gateways/REDIRECTS_FILE.md
- All instances of "implementors" -> "implementers" _(I realize this might be an English dialect thing, just suggesting the American English version because it's commonly used)_

http-gateways/SUBDOMAIN_GATEWAY.md
- reqirements -> requirements, line 157

ipns/IPNS.md
- "implementors" -> "implementers", line 94
- "datastructure" / "datastructures" -> "data structure" / "data structures", line 245

### CODE REVIEW

**Comment by ElPaisano** [comment]
@lidel is this good to merge? Not sure what else needs to be reviewed for spec repo updates

**Review by DannyS03 — APPROVED** [review]

**Review by BigLep — APPROVED** [review]

**Comment by ElPaisano** [comment]
@lidel sorry to message you about this again - still showing that I need your approval to merge this since you're code owner

**Pull request merged**
